pandas_gbq/load.py

```diff
@@ -52,45 +54,110 @@ def split_dataframe(dataframe, chunksize=None):
+    for remaining_rows, chunk in chunks:
+        yield remaining_rows
+
+        client.load_table_from_dataframe(
+            chunk, destination_table_ref, job_config=job_config, location=location,
+        ).result()
+
+
+def load_csv_from_file(
+    client: bigquery.Client,
+    dataframe: pandas.DataFrame,
+    destination_table_ref: bigquery.TableReference,
+    location: Optional[str],
+    chunksize: Optional[int],
+    schema: Optional[Dict[str, Any]],
+):
+    job_config = bigquery.LoadJobConfig()
+    job_config.write_disposition = "WRITE_APPEND"
+    job_config.source_format = "CSV"
+    job_config.allow_quoted_newlines = True
+
+    if schema is None:
         schema = pandas_gbq.schema.generate_bq_schema(dataframe)
-        schema = pandas_gbq.schema.remove_policy_tags(schema)
-        job_config.schema = pandas_gbq.schema.to_google_cloud_bigquery(schema)
+
+    schema = pandas_gbq.schema.remove_policy_tags(schema)
+    job_config.schema = pandas_gbq.schema.to_google_cloud_bigquery(schema)
 
     chunks = split_dataframe(dataframe, chunksize=chunksize)
     for remaining_rows, chunk in chunks:
         yield remaining_rows
 
-        if FEATURES.bigquery_has_from_dataframe_with_csv:
-            client.load_table_from_dataframe(
-                chunk, destination_table_ref, job_config=job_config, location=location,
+        try:
+            chunk_buffer = encode_chunk(chunk)
+            client.load_table_from_file(
+                chunk_buffer,
+                destination_table_ref,
+                job_config=job_config,
+                location=location,
             ).result()
+        finally:
+            chunk_buffer.close()
+
+
+def load_chunks(
+    client,
+    dataframe,
+    destination_table_ref,
+    chunksize=None,
+    schema=None,
+    location=None,
+    api_method="load_parquet",
+):
+    if api_method == "load_parquet":
+        load_parquet(client, dataframe, destination_table_ref, location, schema)
+        # TODO: yield progress depending on result() with timeout
+        return [0]
+    elif api_method == "load_csv":
+        if FEATURES.bigquery_has_from_dataframe_with_csv:
+            return load_csv(
+                client, dataframe, destination_table_ref, location, chunksize, schema
+            )
         else:
-            try:
-                chunk_buffer = encode_chunk(chunk)
-                client.load_table_from_file(
-                    chunk_buffer,
-                    destination_table_ref,
-                    job_config=job_config,
-                    location=location,
-                ).result()
-            finally:
-                chunk_buffer.close()
+            return load_csv_from_file(
+                client, dataframe, destination_table_ref, location, chunksize, schema
+            )
+    else:
+        raise ValueError(
+            f"got unexpected api_method: {api_method!r}, expected one of 'load_parquet', 'load_csv'"
```

Comment:
(nit) Should the first word in the error message be capitalized?

Reply:
Done.

I didn't see anything in PEP-8 or the Google style guide regarding capitalization, but makes sense to me to make error messages have proper capitalization.